test/e2e/framework/control_plane.go

```diff
@@ -84,40 +60,53 @@ func SingleNodeControlPlane(input *SingleNodeControlPlaneInput) {
 	By("creating a Cluster resource linked to the InfrastructureCluster resource")
 	Eventually(func() error {
 		return mgmtClient.Create(ctx, input.Cluster)
-	}, input.CreateTimeout, 10*time.Second).Should(BeNil())
+	}, input.CreateTimeout, eventuallyInterval).Should(Succeed())
+
+	// Create the related resources.
+	By("creating related resources")
+	for _, obj := range input.RelatedResources {
+		By(fmt.Sprintf("creating a/an %s resource", obj.GetObjectKind().GroupVersionKind()))
+		Eventually(func() error {
+			return mgmtClient.Create(ctx, obj)
+		}, input.CreateTimeout, eventuallyInterval).Should(Succeed())
+	}
 
 	// expectedNumberOfNodes is the number of nodes that should be deployed to
-	// the cluster. This is the control plane node and the number of replicas
-	// defined for a possible MachineDeployment.
-	expectedNumberOfNodes := 1
-
-	// Create the control plane machine.
-	By("creating an InfrastructureMachine resource")
-	Expect(mgmtClient.Create(ctx, input.ControlPlaneNode.InfraMachine)).To(Succeed())
-
-	By("creating a BootstrapConfig resource")
-	Expect(mgmtClient.Create(ctx, input.ControlPlaneNode.BootstrapConfig)).To(Succeed())
-
-	By("creating a core Machine resource with a linked InfrastructureMachine and BootstrapConfig")
-	Expect(mgmtClient.Create(ctx, input.ControlPlaneNode.Machine)).To(Succeed())
-
-	By("waiting for cluster to enter the provisioned phase")
-	Eventually(func() (string, error) {
-		cluster := &clusterv1.Cluster{}
-		key := client.ObjectKey{
-			Namespace: input.Cluster.GetNamespace(),
-			Name:      input.Cluster.GetName(),
+	// the cluster. This is the number of control plane nodes and the number of
+	// replicas defined for a possible MachineDeployment.
+	expectedNumberOfNodes := len(input.Nodes)
+
+	// Create the control plane machines.
+	for i, node := range input.Nodes {
+		By(fmt.Sprintf("creating control plane resource %d: InfrastructureMachine", i+1))
+		Expect(mgmtClient.Create(ctx, node.InfraMachine)).To(Succeed())
+
+		By(fmt.Sprintf("creating control plane resource %d: BootstrapConfig", i+1))
+		Expect(mgmtClient.Create(ctx, node.BootstrapConfig)).To(Succeed())
+
+		By(fmt.Sprintf("creating control plane resource %d: Machine", i+1))
+		Expect(mgmtClient.Create(ctx, node.Machine)).To(Succeed())
+
+		// If this is the first node then block until the control plane is
+		// initialized.
+		if i == 0 {
```

Comment:
I’d store `cluster.Status.ControlPlaneInitialized` and test against it here

Reply:
Hi @yastij,

I'm not sure what you mean. That's not how `Eventually` works. What this does is after the first node is deployed, blocks until the control plane is online to deploy additional nodes.